apps/server/src/modules/migration/service/school-migration.service.spec.ts

```diff
@@ -0,0 +1,201 @@
+	});
+
+	const setup = () => {
+		const school: SchoolDO = new SchoolDO({
+			id: 'schoolId',
+			name: 'schoolName',
+			officialSchoolNumber: '3',
+			externalId: 'firstExternalId',
+		});
+		const userDO: UserDO = {
+			id: 'userId',
+			schoolId: school.id as string,
+		} as UserDO;
+		const targetSystemId = 'targetSystemId';
+
+		return {
+			currentUserId: userDO.id as string,
+			officialSchoolNumber: school.officialSchoolNumber,
+			school,
+			externalId: school.externalId as string,
+			userDO,
+			targetSystemId,
+			firstExternalId: school.externalId,
+		};
+	};
+
+	describe('schoolToMigrate is called', () => {
+		describe('when school number is missing', () => {
+			it('should throw an error', async () => {
+				const { currentUserId, externalId } = setup();
+
+				const func = () => service.schoolToMigrate(currentUserId, externalId, undefined);
+
+				await expect(func()).rejects.toThrow(
+					new OAuthMigrationError(
+						'Official school number from target migration system is missing',
+						'ext_official_school_number_missing'
+					)
+				);
+			});
+		});
+
+		describe('when school could not be found with official school number', () => {
+			it('should throw an error', async () => {
+				const { currentUserId, externalId, officialSchoolNumber } = setup();
+				schoolService.getSchoolBySchoolNumber.mockResolvedValue(null);
+
+				const func = () => service.schoolToMigrate(currentUserId, externalId, officialSchoolNumber);
+
+				await expect(func()).rejects.toThrow(
+					new OAuthMigrationError(
+						'Could not find school by official school number from target migration system',
+						'ext_official_school_number_mismatch'
+					)
+				);
+			});
+		});
+
+		describe('when current user is not in the school to migrate to', () => {
+			it('should throw an error', async () => {
+				const { currentUserId, externalId, school, userDO } = setup();
+				schoolService.getSchoolBySchoolNumber.mockResolvedValue(school);
+				userDO.schoolId = 'anotherSchool';
+				userService.findById.mockResolvedValue(userDO);
+
+				const func = () => service.schoolToMigrate(currentUserId, externalId, school.officialSchoolNumber);
+
+				await expect(func()).rejects.toThrow(
+					new OAuthMigrationError(
+						'Current users school is not the same as school found by official school number from target migration system',
+						'ext_official_school_number_mismatch'
+					)
+				);
+			});
+		});
+
+		describe('when school was already migrated', () => {
+			it('should return null ', async () => {
+				const { currentUserId, externalId, school, userDO } = setup();
+				schoolService.getSchoolBySchoolNumber.mockResolvedValue(school);
+				userService.findById.mockResolvedValue(userDO);
+
+				const result: SchoolDO | null = await service.schoolToMigrate(
+					currentUserId,
+					externalId,
+					school.officialSchoolNumber
+				);
+
+				expect(result).toBeNull();
+			});
+		});
+
+		describe('when school was migrated', () => {
```

Comment:
since the other when clause also uses the same tense to describe precondition, this seems inconsistent (you are describing a post-condition here).
Maybe instead write "when school has to be migrated", or "school was not yet migrated"